grpc-proxy/Dockerfile

```diff
@@ -1,15 +1,23 @@
-# Simple dockerfile for the improbable-eng grpc to grpc-web/websocket proxy
+FROM docker.io/library/alpine:3.13
 
-# in theory we can use scratch, but we need the sh -c syntax to include envs vars, or hardcode the backend addr
-FROM alpine
+COPY contents/ .
+
+ENV GRPCWEBPROXY_VERSION=v0.13.0
 
-ADD https://github.com/improbable-eng/grpc-web/releases/download/v0.13.0/grpcwebproxy-v0.13.0-linux-x86_64.zip /grpcwebproxy.zip
 RUN set -eux; \
-    unzip /grpcwebproxy.zip; \
-    mv dist/grpcwebproxy* /grpcwebproxy
+    apk add --no-cache tini; \
+    wget -q "https://github.com/improbable-eng/grpc-web/releases/download/${GRPCWEBPROXY_VERSION}/grpcwebproxy-${GRPCWEBPROXY_VERSION}-linux-x86_64.zip"; \
```

Comment:
wish there was a nicer way to do this... would you consider a follow-up patch that does the download in gradle instead, so we can cache the result there instead of adding to the image?

Reply:
I've looked into it a bit - https://github.com/michel-kraemer/gradle-download-task is a candidate for something that can do it (or, we could roll our own). I prototyped it out a bit, and it's not without its rough edges.

More generally, there's a tradeoff in maintaining caches in gradle vs docker. I see docker as the "easy" but potentially less performant cache; that said, it's easy to grok.

I'm happy to open a ticket for further discussions.

Reply:
#599